runtime/src/main/kotlin/kotlin/native/concurrent/SharedRef.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package kotlin.native.concurrent
+
+import kotlin.native.internal.*
+
+@SymbolName("Kotlin_SharedRef_createSharedRef")
+external private fun createSharedRef(value: Any): NativePtr
+
+/**
+ * A frozen shared reference to a Kotlin object
+ *
+ * Can be safely passed between workers, but can only be accessed on the worker it was created on
+ *
+ * Note: Garbage collector currently cannot detect reference cycles between [SharedRef]s. Consider
```

Comment:
Not exactly between `SharedRef`s, but any reference cycle which has at least one `SharedRef` involved.